### ISSUE TITLE
Check interactions with other TLS 1.3 features

### ISSUE THREAD

**Issue by menonsamir** [chris-wood/draft-bmw-tls-pake13#5]
From IETF discussion with @BjoernMHaase, @FrankXL, and @bifurcation:

We should ensure there aren't unexpected or confusing interactions between our proposal and other TLS features.

The main example brought up so far was resumption tickets. @BjoernMHaase, can you add more detail on your thoughts for those? Thanks!

**Comment by BjoernMHaase** [comment]
Yes. I'd suggest that one digs through all of the subchapters of the full TLS protocol RFC and looks whether we may have to specify something here which may be specific for PAKE protocols.

**Comment by baumanl** [comment]
I read through the 8446 with an eye out for anything affected by the addition of the PAKE extension. Going to close this issue and further discussion can be done on the specific concerns we have.

The main interactions that stood out to me, but that I believe we've already addressed in the draft:

- The PAKE extension provides another key exchange mechanism beyond (EC)DHE, PSK + (EC)DHE, and PSK only. The PAKE provides forward secrecy and replaces the (EC)DHE input to the key schedule.

- The PAKE extension provides another authentication mechanism beyond PSKs + certificate based flows.

- As previously brought up by @davidben and addressed in https://github.com/chris-wood/draft-bmw-tls-pake13/issues/16, the requirements on a valid Client Hello have changed. 

- Hello Retry Requests are not specified as a mechanism to influence the client to offer different PAKESchemes (discussed in https://github.com/chris-wood/draft-bmw-tls-pake13/issues/12)

-  I do not see any major conflicts or points of interest between the pake extension and resumption.  In attempting to resume a client can also provide a "pake" extension to allow fall back to a PAKE auth flow if resumption is not accepted by the server. Like with resumption of a session authenticated with certificates, indefinitely extending the lifetime of the original handshake is probably not a good idea (but 8446 already says this without specifying the auth mechanism specifically). If there is a specific concern here let's open up a separate issue to discuss.


New points I thought of that may warrant changes (most of these are pretty small). I will file separate issues for these:

- We already mention that to prevent client enumeration attacks the server should simulate a response if it does not recognize any of the PAKEShares, but does share a PAKEScheme with the client. It may be good to additionally mention that if the client offers multiple key exchange options (e.g. PSK and pake extension) then the server should have a set preference between those two options instead of say, only using the PSK if none of the PAKEShares are recognized. 

- Do we need to specify any interaction of post handshake auth + the pake handshake? By default I would expect the client sending the post_handshake_auth extension along with the pake extension to be valid and the server sending a CertificateRequest after the handshake has completed. (Not sure what the use case of this would be, but if we don't want to allow it we should specify that in the draft.)

- We recently changed the draft to allow the SPAKE2+ context string to be set by the application instead of hardcoded to a TLS specific value. Should we instead split down the middle and have a hardcoded prefix + application specific context? This may help mitigate cross-protocol attacks.

- We should specify what the client does if verification of confirmV fails.
